### PR TITLE
allow title_image_url param in widget

### PR DESCRIPTION
**Pull request by hejkal**
**NOTE: DO NOT discuss internal CommitChange information in your PR; this PR will be public.
Link back to the issue in the Tix repo when you need to do that.**

Closes commitchange/tix#4389

## example with long narrow logo, and very large (1764 × 1107) title image
Screenshot 2024-10-28 at 10 20 37 AM

## example with very small title image, shorter than the space, showing it centers 
**note that we can't control for inherently uncentered images like the logo in this example, which has slightly more empty space at the bottom than the top of the image.**
Screenshot 2024-10-28 at 10 37 36 AM

### CODE REVIEW

File: client/js/nonprofits/donate/wizard.js [hunk 1]
```diff
@@ -154,6 +154,28 @@ const postDedication = (dedication, donor, donation) => {
   }).load)
 }
 
+const titleInfo = state => {
+  if (state.params$().title_image_url) {
+    return [
+      h('img', {
+        props: {
+          src: state.params$().title_image_url
+        , alt: state.params$().title_image_alt || app.campaign.tagline || app.nonprofit.tagline || ''
```
Comment (hejkal) [inline client/js/nonprofits/donate/wizard.js:163]: should we have a separate param here? should we instruct folks to use the taglines? should we include the designation && single_amount stuff from the existing line below (starting 172)?

File: client/js/nonprofits/donate/wizard.js [hunk 2]
```diff
@@ -154,6 +154,28 @@ const postDedication = (dedication, donor, donation) => {
...
+      }),
+    ];
+  }
+
+  return [
+    h('h2', app.campaign.name || app.nonprofit.name)
+  , h('p', [
+      state.params$().designation && !state.params$().single_amount
+        ? headerDesignation(state)
+        : app.campaign.tagline || app.nonprofit.tagline || '',
```
Comment (wwahammy) [inline client/js/nonprofits/donate/wizard.js:174]: This does bring up an interesting question. Why do we handle this differently if it's a single amount?

File: client/js/nonprofits/donate/wizard.js (hunk 1 above)
Comment (wwahammy) [reply client/js/nonprofits/donate/wizard.js:163]: I think we need a separate param for title_image_alt, there's going to be situations where the title image has nothing to do with the tagline of the campaign or nonprofit.

The question is what we should do if the they choose not to enter an alt. And I don't know.

Comment (wwahammy) [reply client/js/nonprofits/donate/wizard.js:163]: I think we'll go with this as-is in regards to the alt defaults. I don't know of the right way to handle this.

File: client/js/nonprofits/donate/wizard.js (hunk 2 above)
Comment (wwahammy) [reply client/js/nonprofits/donate/wizard.js:174]: We'll answer this question some other time :smile:

File: app/assets/stylesheets/nonprofits/donation_form/title_row.scss [hunk 3]
```diff
@@ -6,9 +6,10 @@
 	position: relative;
 	overflow: hidden;
 	background: $fog;
+	display: flex;
+	gap: 15px;
 }
 .titleRow-info {
```
Comment (wwahammy) [inline app/assets/stylesheets/nonprofits/donation_form/title_row.scss:12]: I think everything should be fine here but I just wanted to make sure you confirmed where these various styles are used throughout the codebase. I think they're only used on the donation form but let's just say I've been surprised before :smile:

**Review by wwahammy — COMMENTED** [review]
Thanks for the PR @hejkal.  I left one comment regarding the styles to address. Additionally, could you share a few screenshots of how this would look with different sized image logo combinations? 

For example, how would it look if:

* the logo is taller than the header image?
* the header image is taller than the logo?
* the header image is very wide?
* the header image is very small?

I just want to make sure we're considering this and we know how it'd work.

File: app/assets/stylesheets/nonprofits/donation_form/title_row.scss (hunk 3 above)
Comment (hejkal) [reply app/assets/stylesheets/nonprofits/donation_form/title_row.scss:12]: @wwahammy Hmm, I am also seeing `titleRow` and `titleRow-info` in `change-amount-wizard.es6`. I'm not sure where to find that in the UI yet to investigate. It does look like it's basically doing the same thing there at first glance.

Comment (hejkal) [reply app/assets/stylesheets/nonprofits/donation_form/title_row.scss:12]: Found that, it appears to look the same as before with this / be rendering essentially the same thing

**Review by wwahammy — APPROVED** [review]

**Pull request merged**